pkg/kafka_client/client.go

```diff
@@ -28,17 +40,47 @@ type KafkaMessage struct {
 }
 
 func NewKafkaClient(options Options) KafkaClient {
-	client := KafkaClient{BootstrapServers: options.BootstrapServers}
+	client := KafkaClient{
+		BootstrapServers: options.BootstrapServers,
+		ConsumerGroupId:  options.ConsumerGroupId,
+		SecurityProtocol: options.SecurityProtocol,
+		SaslMechanisms:   options.SaslMechanisms,
+		SaslUsername:     options.SaslUsername,
+		SaslPassword:     options.SaslPassword,
+		Debug:            options.Debug,
+	}
 	return client
 }
 
 func (client *KafkaClient) consumerInitialize() {
 	var err error
-	client.Consumer, err = kafka.NewConsumer(&kafka.ConfigMap{
+
+	config := kafka.ConfigMap{
 		"bootstrap.servers":  client.BootstrapServers,
 		"group.id":           "kafka-datasource",
 		"enable.auto.commit": "false",
-	})
+	}
+
+	if client.ConsumerGroupId != "" {
+		config.SetKey("group.id", client.ConsumerGroupId)
+	}
+	if client.SecurityProtocol != "" {
+		config.SetKey("security.protocol", client.SecurityProtocol)
+	}
+	if client.SaslMechanisms != "" {
+		config.SetKey("sasl.mechanisms", client.SaslMechanisms)
+	}
+	if client.SaslMechanisms != "" {
+		config.SetKey("sasl.username", client.SaslUsername)
+	}
+	if client.SaslMechanisms != "" {
+		config.SetKey("sasl.password", client.SaslPassword)
+	}
```

Comment:
Only set the option if explicitly configured (values default to "" in the Node code).  This allows use with both PLAINTEXT and SASL_SSL, etc configurations.